### DISCUSSION TITLE
Help in setting up the development setup

### DISCUSSION THREAD

**Discussion by saip007** [google/gcp_scanner#77] [General]
HI Community,
I want to contribute to the project, but unable to setup in local. I want to contibute to GSOC 2023 and want to contribute to "Code Refactoring" project. Help me in starting...

**Comment by peb-peb** [comment]
I guess it's a little late but still for anyone else having the same problem:

Steps to follow for Ubuntu (should be similar across all platforms):
1. Open the Terminal & Clone the project using `git clone https://github.com/google/gcp_scanner.git`
2. Then, 'cd gcp_scanner`
3. Create a Virtual Environment to install all your dependencies with `python3 -m venv venv`
4. Activate the Virtual Environment with `source venv/bin/activate`
5. Install all dependencies using `pip install -r requirements.txt`
6. Install the tool in your local machine `pip install .`
7. Run to check if the tool was installed correctly `gcp-scanner --help`

That's the basic things you need to setup for `gcp-scanner` to develop in local environment.

Some other things to keep in mind before any PR:
1. Check for linting using PyLint:
    for first time do - `wget https://google.github.io/styleguide/pylintrc`
    and then `pylint --rcfile pylintrc --disable=W0703,R1734,R1735,C0209,C0103,R1732 src/gcp_scanner/*.py`
2. Ensure that the tests are successful
3. If any new features have been added, then check with GCP to ensure they work as expected.

P.S. You are required to have a GCP account for development (As of now)

**Reply by mshudrak** [reply]
Thanks for creating that!

**Reply by peb-peb** [reply]
Hey! was working on README. Should we include this in the README?

**Reply by mshudrak** [reply]
It would be better to have a dedicated Wiki page on that.

**Reply by peb-peb** [reply]
ok :+1:
